src/app/static/src/app/components/plots/choropleth/Choropleth.vue

```diff
@@ -0,0 +1,141 @@
+import { Feature } from "geojson";
+import {
+    getVisibleFeatures,
+    getFeatureData,
+    initialiseScaleFromMetadata,
+} from "./utils";
+import ResetMap from "./ResetMap.vue";
+import MapLegend from "./MapLegend.vue";
+import {getColourRange, getScaleLevels} from "../utils";
+import {IndicatorValuesDict, NumericRange} from "../../../types";
+import {ChoroplethIndicatorMetadata} from "../../../generated";
+import { ScaleSettings } from "../../../store/plotState/plotState";
+import {useMapTooltips} from "../useMapTooltips";
+import {useUpdateScale} from "../useUpdateScale";
+
+const store = useStore<RootState>();
+const plotData = computed<PlotData>(() => store.state.plotData.choropleth);
+
+const {updateOutputColourScale} = useUpdateScale();
+const indicatorMetadata = ref<ChoroplethIndicatorMetadata>(store.getters["modelCalibrate/indicatorMetadata"]);
+const colourRange = ref<NumericRange | null>(null);
+const scaleLevels = ref<any>(null);
+const selectedScale = ref<ScaleSettings | null>(null);
+
+const features = store.state.baseline.shape ?
+    store.state.baseline.shape.data.features as Feature[] : [] as Feature[];
+const currentFeatures = ref<Feature[]>([]);
+const featureData = ref<IndicatorValuesDict>({});
+
+const map = ref<typeof LMap | null>(null);
+const featureRefs = ref<typeof LGeoJson[]>([]);
+
+const {createTooltips, updateTooltips} = useMapTooltips(featureData, indicatorMetadata, currentFeatures, featureRefs)
+
+const updateMap = () => {
+    updateFeatures();
+    updateColourScales();
+    updateTooltips();
+};
+
+const colourScales = computed(() => {
+    return store.state.plotState.output.colourScales
+});
+
+const updateColourScales = () => {
+    const colourScales = store.state.plotState.output.colourScales;
+    const selectedIndicator =  store.getters["plotSelections/selectedIndicator"];
+    selectedScale.value = colourScales[selectedIndicator];
+    if (!selectedScale.value) {
+        selectedScale.value = initialiseScaleFromMetadata(indicatorMetadata.value);
+        updateOutputColourScale(selectedScale.value);
+    }
+    colourRange.value = getColourRange(indicatorMetadata.value, selectedScale.value, plotData.value);
+    scaleLevels.value = getScaleLevels(indicatorMetadata.value, colourRange.value);
+    featureData.value = getFeatureData(
+            plotData.value,
+            indicatorMetadata.value,
+            colourRange.value ? colourRange.value : {max: 1, min: 0}
+    );
+}
+
+const updateFeatures = () => {
+    indicatorMetadata.value = store.getters["modelCalibrate/indicatorMetadata"];
+    const selectedLevel = store.state.plotSelections.choropleth.filters
+            .find(f => f.stateFilterId === "detail")!.selection;
+    const selectedAreas = store.state.plotSelections.choropleth.filters
+            .find(f => f.stateFilterId === "area")!.selection;
+    currentFeatures.value = getVisibleFeatures(features, selectedLevel, selectedAreas);
```

Comment:
Do we want to hide areas here? This is not what the current UI is doing, it is just greying out unselected regions